reminder_batch_scheduler/reminder_batch.py

```diff
@@ -0,0 +1,168 @@
+        return
+
+    final_batch = list(selected_batches.keys())[-1]
+    total_cases = sum(selected_batches.values())
+    print()
+    print(f'Selected print batches: {starting_batch} - {final_batch}')
+    print('Current total cases:', total_cases)
+    print('Classifiers JSON for each action type:')
+    for action_type, action_type_classifiers in action_rule_classifiers.items():
+        print()
+        print(f'Action type: {action_type}')
+        print('Classifiers JSON:')
+        print(action_type_classifiers)
+
+    if insert_rules:
+        action_rules = generate_action_rules(action_rule_classifiers, action_plan_id)
+        print()
+        print('Generated action rules:')
+        for action_rule in action_rules.values():
+            print(action_rule)
+        print('')
+        print('About to insert action rules for:')
+        print('Reminder wave:', colored(wave, "red"))
+        print('Action types:', colored(constants.ACTION_TYPES_FOR_WAVE[wave], "red"))
+        print('Print batches:', colored(f'{starting_batch} - {final_batch}', 'red'))
+        print('Current total cases:', colored(total_cases, "red"))
+        if not confirm_insert_rules():
+            print(colored('ABORTING', 'red'))
+            return
+        insert_action_rules(action_rules)
+        print()
+        print(colored("All action rules inserted", 'green'))
+
+
+def count_batch_cases(batch, wave_classifiers):
+    wave_classifiers['print_batch'] = [str(batch)]
+
+    batch_count_query, query_values = build_batch_count_query(wave_classifiers)
+    result = db_helper.execute_parametrized_sql_query(batch_count_query, query_values)
+    return result[0][0]
+
+
+def build_batch_count_query(wave_classifiers):
+    classifier_sql_filters = ['']
+    query_param_values = []
+    for classifier, values in wave_classifiers.items():
+        classifier_sql_filters.append(f" {classifier} IN %s")
+        query_param_values.append(tuple(values))
+    classifiers_query_filters = ' AND'.join(classifier_sql_filters)
+
+    return ("SELECT COUNT(*) FROM actionv2.cases "
+            "WHERE receipt_received = 'f' "
+            "AND address_invalid = 'f' "
+            "AND skeleton = 'f' "
+            "AND refusal_received IS DISTINCT FROM 'EXTRAORDINARY_REFUSAL'"
+            "AND case_type != 'HI'"
+            f"{classifiers_query_filters};"), tuple(query_param_values)
+
+
+def build_action_rule_classifiers(wave, selected_batches):
+    action_rule_classifiers = {}
+    for action_type in constants.ACTION_TYPES_FOR_WAVE[wave]:
+        classifiers = constants.ACTION_TYPE_CLASSIFIERS[action_type]
+        classifiers['print_batch'] = selected_batches
+        action_rule_classifiers[action_type] = json.dumps(classifiers)
+    return action_rule_classifiers
+
+
+def select_batches(starting_batch, wave_classifiers, max_cases):
+    total_cases = 0
+    selected_batches = {}
+
+    for batch in range(starting_batch, 100):
```

Comment:
maximum batch number is supposed to be 99

Reply:
`range` is not inclusive on the end value